connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaStatusBackingStore.java

```diff
@@ -199,14 +199,15 @@ private void sendTaskStatus(final TaskStatus status, boolean safeWrite) {
                                                  final CacheEntry<V> entry,
                                                  final boolean safeWrite) {
         final int sequence;
+        final byte[] value;
         synchronized (this) {
             this.generation = status.generation();
             if (safeWrite && !entry.canWriteSafely(status))
                 return;
             sequence = entry.increment();
-        }
 
-        final byte[] value = status.state() == ConnectorStatus.State.DESTROYED ? null : serialize(status);
+            value = status.state() == ConnectorStatus.State.DESTROYED ? null : serialize(status);
```

Comment:
The call `serialize(status)` calls a `converter.fromConnectData` inside. Per our discussion regarding the converter I'll revert this one too.